tap_yotpo/streams/product_variants.py

```diff
@@ -0,0 +1,127 @@
+    api_auth_version = ApiSpec.API_V3
+    # points to the attribute of the config that marks the first-start-date for the stream
+    config_start_key = "start_date"
+    url_endpoint = "https://api.yotpo.com/core/v3/stores/APP_KEY/products/PRODUCT_ID/variants"
+
+    def __init__(self, client=None) -> None:
+        super().__init__(client)
+        self.sync_prod: bool = True
+        self.last_synced: bool = False
+        self.base_url = self.get_url_endpoint()
+
+    def get_products(self, state) -> Tuple[List, int]:
+        """Returns index for sync resuming on interruption."""
+        shared_product_ids = Products(self.client).prefetch_product_ids()
+        last_synced = get_bookmark(state, self.tap_stream_id, "currently_syncing", False)
+        last_sync_index = 0
+        if last_synced:
+            for pos, (prod_id, _) in enumerate(shared_product_ids):
+                if prod_id == last_synced:
+                    LOGGER.warning("Last Sync was interrupted after product *****%s", str(prod_id)[-4:])
+                    last_sync_index = pos
+                    break
+        return shared_product_ids, last_sync_index
+
+    def get_records(self, prod_id: str, bookmark_date: str) -> Tuple[List, datetime]:
+        # pylint: disable=W0221
+        """Performs api querying and pagination of response.
+
+        Retrieves all record and filters within the code, as the API
+        endpoint does not have any query parameter to fetch the latest
+        record from specific date.
+        """
+        extraction_url = self.base_url.replace("PRODUCT_ID", prod_id)
+        bookmark_date = current_max = strptime_to_utc(bookmark_date)
+        filtered_records = []
+        page_count, params = 1, {}
+        while True:
+            LOGGER.info("Calling Page %s", page_count)
+
+            response = self.client.get(extraction_url, params, {}, self.api_auth_version)
+
+            # response = response.get("response", {})
+            raw_records = response.get("variants", [])
+            pagination = response.get("pagination", {}).get("next_page_info", None)
+
+            if not raw_records:
+                break
+
+            for record in raw_records:
+                record_timestamp = strptime_to_utc(record[self.replication_key])
+                if record_timestamp >= bookmark_date:
+                    current_max = max(current_max, record_timestamp)
+
+                    # Adding yotpo_product_id in record
+                    if "yotpo_product_id" not in record.keys():
+                        record["yotpo_product_id"] = int(prod_id)
+                    filtered_records.append(record)
+
+            if not pagination:
+                break
+            else:
+                params["page_info"] = pagination
+            page_count += 1
+
+        return (filtered_records, current_max)
+
+    def sync(self, state: Dict, schema: Dict, stream_metadata: Dict, transformer: Transformer) -> Dict:
```

Comment:
The sync method is overridden in all the classes, so how about putting a more useful sync method in the Incremental class instead of the standard one? this will lead to less code.

Reply:
We have already done a default implementation, will review it and check if this can be improved,
https://github.com/singer-io/tap-yotpo/blob/ee90232f67deb53a69abdc86c7ef367e8ea6758d/tap_yotpo/streams/abstracts.py#L131-L172

if you have any suggestions on how this can be improved, please lmk.

Reply:
Yes, I checked that and that looks good to me but have we used that in any class because I can see Sync method in almost all stream classes.